src/kvstore/EventListner.h

```diff
@@ -13,31 +13,243 @@ namespace kvstore {
+                return "FIFOReduceNumFiles";
+            case rocksdb::CompactionReason::kFIFOTtl:
+                return "FIFOTtl";
+            case rocksdb::CompactionReason::kManualCompaction:
+                return "ManualCompaction";
+            case rocksdb::CompactionReason::kFilesMarkedForCompaction:
+                return "FilesMarkedForCompaction";
+            case rocksdb::CompactionReason::kBottommostFiles:
+                return "BottommostFiles";
+            case rocksdb::CompactionReason::kTtl:
+                return "Ttl";
+            case rocksdb::CompactionReason::kFlush:
+                return "Flush";
+            case rocksdb::CompactionReason::kExternalSstIngestion:
+                return "ExternalSstIngestion";
+            case rocksdb::CompactionReason::kPeriodicCompaction:
+                return "PeriodicCompaction";
+            case rocksdb::CompactionReason::kNumOfReasons:
+                return "NumOfReasons";
+        }
+        return "Unknown";
+    }
+
+    std::string flushReasonString(const rocksdb::FlushReason& reason) {
+        switch (reason) {
+            case rocksdb::FlushReason::kOthers:
+                return "Others";
+            case rocksdb::FlushReason::kGetLiveFiles:
+                return "GetLiveFiles";
+            case rocksdb::FlushReason::kShutDown:
+                return "ShutDown";
+            case rocksdb::FlushReason::kExternalFileIngestion:
+                return "ExternalFileIngestion";
+            case rocksdb::FlushReason::kManualCompaction:
+                return "ManualCompaction";
+            case rocksdb::FlushReason::kWriteBufferManager:
+                return "WriteBufferManager";
+            case rocksdb::FlushReason::kWriteBufferFull:
+                return "WriteBufferFull";
+            case rocksdb::FlushReason::kTest:
+                return "Test";
+            case rocksdb::FlushReason::kDeleteFiles:
+                return "DeleteFiles";
+            case rocksdb::FlushReason::kAutoCompaction:
+                return "AutoCompaction";
+            case rocksdb::FlushReason::kManualFlush:
+                return "ManualFlush";
+            case rocksdb::FlushReason::kErrorRecovery:
+                return "ErrorRecovery";
+        }
+        return "Unknown";
+    }
+
+    std::string backgroundErrorReasonString(const rocksdb::BackgroundErrorReason& reason) {
+        switch (reason) {
+            case rocksdb::BackgroundErrorReason::kFlush:
+                return "Flush";
+            case rocksdb::BackgroundErrorReason::kCompaction:
+                return "Compaction";
+            case rocksdb::BackgroundErrorReason::kWriteCallback:
+                return "WriteCallback";
+            case rocksdb::BackgroundErrorReason::kMemTable:
+                return "MemTable";
+        }
+        return "Unknown";
+    }
+
+    std::string writeStallConditionString(const rocksdb::WriteStallCondition& condition) {
```

Comment:
Print the prev and cur condition in OnStallConditionsChanged

Reply:
fixed